salt/cloud/clouds/gce.py

```diff
@@ -387,18 +390,39 @@ def __get_size(conn, vm_):
     """
     Need to override libcloud to find the machine type in the proper zone.
     """
+    default_size = "n1-standard-1"
     size = config.get_cloud_config_value(
-        "size", vm_, __opts__, default="n1-standard-1", search_global=False
+        "size", vm_, __opts__, default=default_size, search_global=False
     )
+    if not size or not isinstance(size, str):
+        size = default_size
     return conn.ex_get_size(size, __get_location(conn, vm_))
 
 
+def __get_labels(vm_):
+    """
+    Get configured labels.
+    """
+    l = config.get_cloud_config_value(
+        "ex_labels", vm_, __opts__, default="{}", search_global=False
+    )
+    # Consider warning the user that the labels in the cloud profile
+    # could not be interpreted, bad formatting?
+    try:
+        labels = literal_eval(l)
+    except Exception:  # pylint: disable=W0703
+        labels = None
+    if not labels or not isinstance(labels, dict):
+        labels = None
+    return labels
+
+
 def __get_tags(vm_):
     """
     Get configured tags.
     """
     t = config.get_cloud_config_value(
-        "tags", vm_, __opts__, default="[]", search_global=False
+        "ex_tags", vm_, __opts__, default="[]", search_global=False
```

Comment:
why does this need to be changed to `ex_`?

Reply:
Right - this was changed originally because the test config originally provided in test_gce.py provided `ex_tags`.
https://github.com/saltstack/salt/pull/62046/files/5cd5fd75bb7eb7a60e6d73d4efc7611e482ebff9#diff-011c0910037d4cbee4d337f9311903470b81753065ce3586eff691d536f5a465L69

This isn't correct though, `tags` the is the right key. Have put this back and corrected the test config